pandas/tests/arrays/categorical/test_constructors.py

```diff
@@ -149,6 +149,17 @@ def test_constructor(self):
                          categories=["a", "b", "c", "d"])
         tm.assert_categorical_equal(c1, c2)
 
+        # GH25318: constructing with pd.Series used to bogusly skip recoding
```

Comment:
make this added section a separate test name: test_constructor_with_existing_categories